fceux: update and adopt

AndersonTorres opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Closes https://github.com/NixOS/nixpkgs/pull/282415

Many thanks to @fmahnke

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

AndersonTorres committed:
fceux: 2.6.5 -> 2.6.6

Co-authored-by: Fritz Mahnke <fritz@mahnke.tech>

AndersonTorres committed:
fceux: unpin lua and stdenv

AndersonTorres committed:
fceux: 2.6.6 -> 2.6.6-unstable-2024-01-19

Since 2.6.6 is from August 2023.

AndersonTorres committed:
fceux: 2.6.4 -> 2.6.5

Also, pin stdenv to GCC12 - yet another case of updated incompatible C++
standards I suppose.

Oh! Remove it when next version arrives ;)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3308

SuperSandro2000 commented:
Is there any other motivation for the unstable update other than that the last release is from August 2023?

AndersonTorres commented:
> Is there any other motivation for the unstable update other than that the last release is from August 2023?

Yes and no. 
My main motivation was/is to catch HEAD, indeed.

On the other hand I found at least one serious update: a memleak caused by not closing file handlers
https://github.com/TASEmulators/fceux/commit/d86c81e26effc4e401bb12ba2d871f02284e97f2

There is also updates for QT6 - but I don't know how to make this choice between QTs a smooth and elegant experience.

trofi commented:
Result of `nixpkgs-review pr 282424` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fceux</li>
  </ul>
</details>

trofi commented:
Also fixes build failure happening on `master` due to `gcc-13` update: https://hydra.nixos.org/build/247726311

AndersonTorres committed:
fceux: update meta

- get rid of nested with
- add AndersonTorres to maintainers
- remove scubed2
- add changelog

AndersonTorres commented on pkgs/applications/emulators/fceux/default.nix:
```diff
@@ -34,11 +34,12 @@ stdenv.mkDerivation (finalAttrs: {
-    platforms = platforms.linux;
+    changelog = "https://raw.githubusercontent.com/TASEmulators/fceux/${finalAttrs.src.rev}/changelog.txt";
+    license = with lib.licenses; [ gpl2Plus ];
+    maintainers = with lib.maintainers; [ AndersonTorres sbruder scubed2 ];
```
Solved
